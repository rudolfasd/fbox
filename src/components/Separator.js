import React from 'react';


class Separator extends React.Component {

    sepBtnClasses = () => {
        let classes = ['sep-btn-del-all'];
        let order = this.props.pointList.order;
        if (order.length === 0){
            classes.push('inactive');
        };
        return classes.join(' ');
    };
    
    delAllPoints = () => {
        let order = this.props.pointList.order;
        if (order.length){
            this.props.sagaDelAllPoints();
        };
    };

    render(){
        return(
            <div className='sep' onClick={this.delAllPoints}>
                <div className='sep-line'></div>
                <div className={this.sepBtnClasses()}></div>
            </div>
        );
    };
};

export default Separator;