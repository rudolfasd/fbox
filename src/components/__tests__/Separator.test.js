import React from 'react';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Separator from '../Separator';


describe('REACT COMPONENT <Separator />',()=>{
    const mockSagaDelAllPoints = jest.fn();
    const mockInitProps = {
        pointList: { order: []},
        sagaDelAllPoints: mockSagaDelAllPoints
    };
    const mockProps = { 
        pointList: {order: [2,5,6,8]},
        sagaDelAllPoints: mockSagaDelAllPoints
    };
    let renderedValue, component;

    it('Separator - snapshot if no points',() => {
        renderedValue = renderer.create(
            <Separator {...mockInitProps}/>
        ).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });
    it('Separator - snapshot if points exist',() => {
        renderedValue = renderer.create(
            <Separator {...mockProps}/>
        ).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });

    it('Separator - sepBtnClasses if no points',()=>{
        component = shallow(<Separator {...mockInitProps} />);
        expect(component.instance().sepBtnClasses())
            .toEqual('sep-btn-del-all inactive');
    });
    it('Separator - sepBtnClasses if points exist',()=>{
        component = shallow(<Separator {...mockProps} />);
        expect(component.instance().sepBtnClasses())
            .toEqual('sep-btn-del-all');
    });

    it('Separator - sagaDelAllPoints if no points',()=>{
        component = shallow(<Separator {...mockInitProps} />);
        component.instance().delAllPoints();
        expect(mockSagaDelAllPoints).toHaveBeenCalledTimes(0);
    });
    it('Separator - sagaDelAllPoints if points exist',()=>{
        component = shallow(<Separator {...mockProps} />);
        component.instance().delAllPoints();
        expect(mockSagaDelAllPoints).toHaveBeenCalledTimes(1);
    });
});