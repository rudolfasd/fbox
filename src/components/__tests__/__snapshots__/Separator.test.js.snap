// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REACT COMPONENT <Separator /> Separator - snapshot if no points 1`] = `
<div
  className="sep"
  onClick={[Function]}
>
  <div
    className="sep-line"
  />
  <div
    className="sep-btn-del-all inactive"
  />
</div>
`;

exports[`REACT COMPONENT <Separator /> Separator - snapshot if points exist 1`] = `
<div
  className="sep"
  onClick={[Function]}
>
  <div
    className="sep-line"
  />
  <div
    className="sep-btn-del-all"
  />
</div>
`;
